core/templates/dev/head/pages/library/Library.js

```diff
@@ -185,28 +219,33 @@ oppia.controller('Library', [
       $scope.$apply();
     });
 
-    // The following checks if the page is in search mode.
-    $scope.inSearchMode = ($window.location.pathname.indexOf('/search') === 0);
     var activateSearchMode = function() {
-      if (!$scope.inSearchMode) {
+      if ($scope.pageMode !== LIBRARY_PAGE_MODES.SEARCH) {
         $('.oppia-library-container').fadeOut(function() {
-          $scope.inSearchMode = true;
+          $scope.pageMode = LIBRARY_PAGE_MODES.SEARCH;
           $timeout(function() {
             $('.oppia-library-container').fadeIn();
           }, 50);
         });
       }
     };
 
-    $scope.showAllResultsForCategories = function(categories) {
-      var selectedCategories = {};
-      for (i = 0; i < categories.length; i++) {
-        selectedCategories[categories[i]] = true;
-      }
+    // The following loads explorations belonging to a particular group. If
+    // fullResultsUrl is given it loads the page corresponding to the url. If
+    // not given it will initiate a search query based on categories.
```

Comment:
"If not given " --> "Otherwise, "
"based on categories" --> "for the given list of categories."